my-react-study-app/src/1_app/app.tsx

```diff
@@ -0,0 +1,18 @@
+import { Route, Routes } from 'react-router-dom';
+import './app.css';
+import BasicPage from '../3_pages/basicPage';
+import NotFoundPage from '../3_pages/notFoundPage/notFoundPage';
+import DetailedItem from '../4_widgets/detailedItem/detailedItem';
+
+export default function App() {
+    return (
+        <Routes>
+            <Route path="/" element={<BasicPage />}>
+                <Route path="character/detail/:id" element={<DetailedItem />} />
+                <Route path="episode/detail/:id" element={<DetailedItem />} />
+                <Route path="location/detail/:id" element={<DetailedItem />} />
```

Comment:
it's strange to use one component for different routes, try to use different components. We can discuss it